build/builder_test.go

```diff
@@ -230,3 +234,268 @@ func TestDontCountContentOfSkippedFiles(t *testing.T) {
 		t.Fatalf("content of skipped documents should not count towards shard size thresold")
 	}
 }
+
+func TestOptions_FindAllShards(t *testing.T) {
+	type normalShard struct {
```

Comment:
nit: we use "simple shard" in the code base already, maybe rename normalShard for conistency? I am guessing you refer to shards that only contain 1 repo?